### PR TITLE
optimizer-hints: add docs for `nth_plan()` hint

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)
Add dos for `nth_plan()`hint in optimizer-hints.md.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4227
- Other reference link(s): https://github.com/pingcap/tidb/issues/17580

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: optimizer-hints.md [hunk 1]
```diff
@@ -313,3 +313,23 @@ In the following example, the Plan Cache is forcibly disabled when executing the
 ```sql
 prepare stmt from 'select  /*+ IGNORE_PLAN_CACHE() */ * from t where t.id = ?';
 ```
+
+### NTH_PLAN(N)
+
+The `NTH_PLAN(N)` hint reminds the optimizer to select the `N`th physical plan found in the physical optimization stage. `N` must be a positive integer.
```
Comment (TomShawn) [inline optimizer-hints.md:319]:
```suggestion
The `NTH_PLAN(N)` hint reminds the optimizer to select the `N`th physical plan found during the physical optimization. `N` must be a positive integer.
```

File: optimizer-hints.md [hunk 2]
```diff
@@ -313,3 +313,23 @@ In the following example, the Plan Cache is forcibly disabled when executing the
...
+
+If the specified `N` exceeds the search range of the physical optimization stage, TiDB will return a warning and select the optimal physical plan based on the same strategy as when this hint does not exist.
```
Comment (TomShawn) [inline optimizer-hints.md:321]:
```suggestion
If the specified `N` is beyond the search range of the physical optimization, TiDB will return a warning and select the optimal physical plan based on the strategy that ignores this hint.
```

File: optimizer-hints.md [hunk 3]
```diff
@@ -313,3 +313,23 @@ In the following example, the Plan Cache is forcibly disabled when executing the
...
+
+This hint does not take effect when the cascades planner is enabled.
+
+In the following example, the optimizer is forcibly selected the third physical plan found in the physical optimization stage:
```
Comment (TomShawn) [inline optimizer-hints.md:325]:
```suggestion
In the following example, the optimizer is forced to select the third physical plan found during the physical optimization:
```

File: optimizer-hints.md [hunk 4]
```diff
@@ -313,3 +313,23 @@ In the following example, the Plan Cache is forcibly disabled when executing the
...
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT /*+ NTH_PLAN(3) */ count(*) from t where a > 5;
+```
+
+> **Note:**
+>
+> `NTH_PLAN(N)` is mainly used for testing, and its compatibility is not guaranteed in later versions, so proceed **with caution**.
```
Comment (TomShawn) [inline optimizer-hints.md:335]:
```suggestion
> `NTH_PLAN(N)` is mainly used for testing, and its compatibility is not guaranteed in later versions. Use this hint **with caution**.
```

**Review by TomShawn — COMMENTED** [review]
Rest LGTM

**Review by time-and-fate — COMMENTED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**